api/v1alpha1/storageconsumer_types.go

```diff
@@ -42,7 +43,23 @@ type StorageConsumerSpec struct {
 	Enable bool `json:"enable,omitempty"`
 	// StorageQuotaInGiB describes quota for the consumer
 	// +optional
-	StorageQuotaInGiB int `json:"storageQuotaInGiB,omitempty"`
+	StorageQuotaInGiB          int                            `json:"storageQuotaInGiB,omitempty"`
+	CephResourcesConfigMapRef  corev1.LocalObjectReference    `json:"cephResourcesConfigMapRef,omitempty"`
+	StorageClasses             []StorageClassSpec             `json:"storageClasses,omitempty"`
+	VolumeSnapshotClasses      []VolumeSnapshotClassSpec      `json:"volumeSnapshotClasses,omitempty"`
+	VolumeGroupSnapshotClasses []VolumeGroupSnapshotClassSpec `json:"volumeGroupSnapshotClasses,omitempty"`
```

Comment:
It'd be helpful if these fields

- supported patch operations, so that users can patch individual fields rather than updating the entire spec.
- supported ServerSideApply to build automation.
- included validations to ensure that array items are unique.
- provided labelSelector instead of name, to distribute a group of resources.

These could be discussed and added in a separate PR, but I think these would be useful features that would enable many usecases. @nb-ohad thoughts?

Reply:
> supported patch operations
- if you mean by SMP I don't think CRDs can do that (only Aggregated APIs) iirc. Merge & JsonPatch is supported by default anyways

> ServerSideApply to build automation.
- good one, we could certainly do that

> included validations
- yes, we want to and mostly in a different PR

> provided labelSelector instead of name,
- iirc, this was discussed in one of the meetings and wasn't chosen as this creates a two step operations and the number of SCs would be in single digits, open for any discussions though

thanks for the suggestions! will await Ohad's reply as well.

Reply:
@umangachapagain Can you please create an RFE issue on this repo for each feature you are interested in with justification and a use case?


> provided labelSelector instead of name,

This is undesired, we don't want to group distribution of multiple entities under a single key. We intentionally decided to use a name as it can only identify a single CR at a single point in time (as opposed to UUID which identify a fixed single CR, or a selector which can identify a group of CRs at a single point in time)

Reply:
"we don't want to group distribution of multiple entities under a single key". Why not?

I am not requesting for grouping of multiple types under same label set. I am asking for grouping of resources of same type that needs to be distributed to same client. Why should I be forced to name each one explicitly?
"Distribute all StorageClasses with LabelA to ClientA. Distribute all StorageClasses with LabelB to ClientB. " seems like a reasonable request.

Reply:
I am not saying it will not be handy, but at the same time it does not serve any of agreed UX designs. The current UX is build upon selecting of individual entities upon creation of the entity or upon creation of the consumer (we support both)

I am not saying we should not consider that as a future enhancement, but I do want a justification that is more then just "it be cool to have such flexibility".  You need to remember that we are talking about a resource space where we anticipate a single-digit count of entities (of each type), not hundreds or thousands of entities (which is the case where labels become an important tool)